### PR TITLE
feat: Infra nginx opala + jwt payload sub support (MAPCO-3998) (MAPCO-4010)

### PR DESCRIPTION
**Pull request by CL-SHLOMIKONCHA**
| Question                | Answer                                                                          |
| ---------------- | -------------------------------------------------------------------------- |
| Bug fix         | ✖                                                                        |
| New feature     | ✔                                                                        |
| Breaking change | ✖                                                                        |
| Deprecations    | ✖                                                                        |
| Documentation   | ✖                                                                        |
| Tests added     | ✔                                                                         |
| Chore            | ✖                                                                       |

### CODE REVIEW

File: helm/config/nginx/nginx.conf [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+
+    sendfile        on;
+    #tcp_nopush     on;
+
+    keepalive_timeout  65;
+
+    gzip  on;
+
+    include /etc/nginx/conf.d/*.conf;
+}
```
Comment (asafMasa) [inline helm/config/nginx/nginx.conf:38]: end of file

File: tests/unit/taskManager/taskManager.spec.ts [hunk 2]
```diff
@@ -100,8 +100,56 @@ describe('taskManager', () => {
         jobId: response.jobId,
         estimatedSize: estimationsResponse.estimatedFileSize,
         estimatedTime: estimationsResponse.estimatedTime,
-        // TODO: handle customer name once implemented
-        customerName: 'cutomer_name',
+      });
+    });
+    it('resolves and call with the received customer name', async () => {
```
Comment (asafMasa) [inline tests/unit/taskManager/taskManager.spec.ts:105]: separate methods with line

File: tests/unit/taskManager/taskManager.spec.ts [hunk 3]
```diff
@@ -100,8 +100,56 @@ describe('taskManager', () => {
...
+
+      await expect(createPromise).resolves.not.toThrow();
+      expect(getEstimationsSpy).toHaveBeenCalledTimes(1);
+      expect(createTask).toHaveBeenCalledTimes(1);
+      expect(createTask).toHaveBeenCalledWith({
+        ...entity,
+        customerName: customerName,
+      });
+    });
+    it('resolves and call with the customer name as undefined', async () => {
```
Comment (asafMasa) [inline tests/unit/taskManager/taskManager.spec.ts:130]: separate methods with line

**Review by asafMasa — CHANGES_REQUESTED** [review]
👍

File: helm/.helmignore [hunk 4]
```diff
@@ -21,3 +21,5 @@
 .idea/
 *.tmproj
 .vscode/
+# Common local deployment files
+local.yaml
```
[the remark this answers is not in the document]
Comment (CL-SHLOMIKONCHA) [reply helm/.helmignore:25]: fixed

File: helm/Chart.yaml [hunk 5]
```diff
@@ -4,3 +4,7 @@ description: A Helm chart for export-management service
 type: application
 version: 1.1.4
 appVersion: 1.1.4
+dependencies:
+  - name: nginx
+    version: 1.1.0
+    repository: oci://acrarolibotnonprod.azurecr.io/helm
```
[the remark this answers is not in the document]
Comment (CL-SHLOMIKONCHA) [reply helm/Chart.yaml:10]: fixed

File: helm/config/nginx/default.conf [hunk 6]
```diff
@@ -0,0 +1,53 @@
...
+    proxy_connect_timeout       600;
+    proxy_send_timeout          600;
+    send_timeout                600;
+    client_max_body_size        5000;
+    client_header_timeout       600;
+    client_body_timeout         600;
+    client_header_buffer_size   12288; # 12K
+    large_client_header_buffers 4 12288; # 12K
+
+    location /liveness {
```
[the remark this answers is not in the document]
Comment (CL-SHLOMIKONCHA) [reply helm/config/nginx/default.conf:24]: fixed

File: helm/values.yaml [hunk 7]
```diff
@@ -90,22 +86,60 @@ opa:
...
+    domain: 'export'
+    url: 'http://localhost:8181/v1/data/http/authz/decision'
+
+  route:
+    enabled: true
+    path: /
+    host: 
+    timeout:
+      enabled: false # defaults to 30s by openshift
+      duration: 60s # supported units (us, ms, s, m, h, d)
```
[the remark this answers is not in the document]
Comment (CL-SHLOMIKONCHA) [reply helm/values.yaml:125]: Roee should answer this.

File: helm/values.yaml [hunk 8]
```diff
@@ -90,22 +86,60 @@ opa:
...
+    tls:
+      enabled: true
+      termination: edge
+      insecureEdgeTerminationPolicy: Redirect
+      useCerts: false
+      certificate: ''
+      key: ''
+      caCertificate: ''
+
+  ingress:
```
[the remark this answers is not in the document]
Comment (CL-SHLOMIKONCHA) [reply helm/values.yaml:136]: fixed

File: helm/config/nginx/nginx.conf (hunk 1 above)
Comment (CL-SHLOMIKONCHA) [reply helm/config/nginx/nginx.conf:38]: fixed

File: tests/unit/taskManager/taskManager.spec.ts (hunk 2 above)
Comment (CL-SHLOMIKONCHA) [reply tests/unit/taskManager/taskManager.spec.ts:105]: fixed

File: tests/unit/taskManager/taskManager.spec.ts (hunk 3 above)
Comment (CL-SHLOMIKONCHA) [reply tests/unit/taskManager/taskManager.spec.ts:130]: fixed

File: helm/templates/nginx/configmap.yaml [hunk 9]
```diff
@@ -0,0 +1,16 @@
...
+  name: {{ .Release.Name }}-{{ $chartName }}-nginx-configmap
+  labels:
+    app: {{ $releaseName }}-{{ $chartName }}
+    component: nginx-configmap
+    environment: {{ .Values.environment }}
+    release: {{ $releaseName }}
+data:
+  default.conf: {{ tpl (.Files.Get "config/nginx/default.conf") . | quote }}
+  nginx.conf: {{ tpl (.Files.Get "config/nginx/nginx.conf") . | quote }}
+{{- end }}
```
[the remark this answers is not in the document]
Comment (CL-SHLOMIKONCHA) [reply helm/templates/nginx/configmap.yaml:16]: fixed

File: helm/values.yaml [hunk 10]
```diff
@@ -90,22 +86,60 @@ opa:
...
-        memory: 128Mi
-      requests:
-        cpu: 100m
-        memory: 128Mi
+    domain: 'export'
+    url: 'http://localhost:8181/v1/data/http/authz/decision'
+
+  route:
+    enabled: true
+    path: /
```
[the remark this answers is not in the document]
Comment (CL-SHLOMIKONCHA) [reply helm/values.yaml:121]: fixed

**Pull request merged**